app/services/resolver_cost_calculator.rb

```diff
@@ -0,0 +1,35 @@
+class ResolverCostCalculator
+  def initialize(source)
+    @source = source
+  end
+
+  def cost
+    @source.outcome == 'none' ? 0 : incurred_cost(@source)
+  end
+
+  private
+
+  def incurred_cost(source)
+    if source.is_a?(PartPayment)
+      if source.application.evidence_check.present?
+        fee(source) - amount_to_pay(source.application.evidence_check)
+      else
+        fee(source) - amount_to_pay(source.application)
+      end
+    else
+      fee(source)
+    end
+  end
+
+  def fee(source)
+    if source.is_a?(Application)
+      source.detail.fee
+    else
+      source.application.detail.fee
+    end
+  end
```

Comment:
The same thing here, maybe it can be re-worked so that `source` is not passed around, but the instance variable used instead?